stubs/parsimonious/parsimonious/nodes.pyi

```diff
@@ -0,0 +1,40 @@
+from re import Match
+from typing import Any, Callable, Iterator, NoReturn, TypeVar
+from typing_extensions import Self
```

Comment:
We can't use PEP 673 yet because mypy doesn't support it

Reply:
good shout, I've removed the function it was used in anyway 👍